Source/WebCore/rendering/RenderLayerModelObject.cpp

```diff
@@ -158,7 +158,7 @@ void RenderLayerModelObject::styleDidChange(StyleDifference diff, const RenderSt
             if (s_wasFloating && isFloating())
                 setChildNeedsLayout();
             createLayer();
-            if (parent() && !needsLayout() && containingBlock())
+            if (parent() && !needsLayout())
```

Comment:
Well, the question is whether we expect a parent or a containing block here? Let me take a look.

Reply:
Both checks were added 20 years ago containingBlock() -> https://commits.webkit.org/7750@main followed by parent() -> https://commits.webkit.org/8988@main but they were introduced to "protect" `updateLayerPosition` which is not even there anymore.
Since at this point all we wanna do is set the repaint status on the layer, probably neither of this is needed and instead we could just go with the `needsLayout()` check....but let's just remove the redundant tree traversal for now.